openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfileManager.java

```diff
@@ -409,26 +410,49 @@ public RouteResult matchTrack(MapMatchingRequest req) throws Exception {
                 }
             }
 
-            try {
-                for (Object obj : gr.getReturnObjects()) {
-                    if (obj instanceof ExtraInfoProcessor) {
-                        if (extraInfoProcessor == null) {
-                            extraInfoProcessor = (ExtraInfoProcessor)obj;
+            List<ExtraInfoProcessor> extraList = new ArrayList<>();
+            for (Object o : gr.getReturnObjects()) {
+                if (o instanceof ExtraInfoProcessor) {
+                    extraList.add((ExtraInfoProcessor)o);
+                }
+            }
+
+            if (numberOfExpectedExtraInfoProcessors > 1) {
+                int extraInfoProcessorIndex = 0;
+                for (Object o : gr.getReturnObjects()) {
+                    if (o instanceof ExtraInfoProcessor) {
+                        extraInfoProcessor[extraInfoProcessorIndex] = (ExtraInfoProcessor)o;
+                        extraInfoProcessorIndex++;
+                    }
+                }
+            } else {
```

Comment:
Why is it necessary to distinguish between more-than-one and one-or-less expected extraInfoProcessors?

Reply:
because multiple EIPs could mean two different things: either multiple paths are separate routes with an EIP each, or they are segments of one route and the EIPs need to be concatenated.